### DISCUSSION TITLE
How to custom table layout?

### DISCUSSION THREAD

**Discussion by sunxyw** [filamentphp/filament#1851] [Help] [answered]
Hi, is there any possible way to allow us to modify the table layout?
Like make a table with a list of cards, eg:
![圖片](https://user-images.githubusercontent.com/31698606/157601011-dcb388e7-7b49-4b37-ab5a-a67bb7be18b3.png)

I know it can be achieved using custom pages, but is there a way to integrate with the table builder? Allows changing the layout of the table without changing the specific code. For example, simply setting the `view` field?

I noticed that there is a field called `$view` in the `ListRecords` class.
`protected static string $view = 'filament::resources.pages.list-records';`

I found that the rendering of the table behaves as `{{ $this->table }}` in the view, and I'm guessing it comes from [`packages/tables/src/Table.php`](https://github.com/laravel-filament/filament/blob/HEAD/packages/tables/src/Table.php#L334), which pass the `table` variable to the view.

But I'm stuck here, I can't find where exactly the table view is.

I think it should be [`packages/tables/resources/views/components/table.blade.php`](https://github.com/laravel-filament/filament/blob/503a0a588f/packages/tables/resources/views/components/table.blade.php) but I can't find the connection between it and the code.

Can someone tell me how should I change the view of the table?

Thank you.

**Comment by studiowizjo** [comment]
Inspect this file: [\vendor\filament\tables\resources\views\index.blade.php](https://github.com/laravel-filament/filament/blob/503a0a588fb1094dbb24ae22b2ead73e2bfc2b64/packages/tables/resources/views/index.blade.php). 

You can see there how table component is called below the line:

```php
@if (($records = $getRecords())->count())
```

**Comment by danharrin** [comment, marked as answer]
There's actually a `getTableContent()` method now, and you can return a view for the table cards there. You can access the `$records` in there. Overriding views is not reccomended :)

**Reply by aurawindsurfing** [reply]
@danharrin I have a problem figuring that out in 3.0 was the method removed? Can you give an example how to achieve that? Thanks!

**Reply by danharrin** [reply]
https://filamentphp.com/docs/3.x/tables/layout#arranging-records-into-a-grid

**Reply by aurawindsurfing** [reply]
Got it! Thank you so much. I anded up simply creating View::make for a card I wanted.

**Reply by aurawindsurfing** [reply]
BTW how would you get rid of the default cell padding in the table layout? Custom theme?

**Reply by danharrin** [reply]
Yeah you can use CSS in a custom theme

**Comment by aurawindsurfing** [comment]
Actually @sunxyw it got improved and now you have grids available: https://filamentphp.com/docs/2.x/tables/layout#arranging-records-into-a-grid

**Reply by sourabhsurana1008** [reply]
It is not working

**Comment by shebaoting** [comment]
@sunxyw https://github.com/filamentphp/filament/discussions/1851

**Reply by sunxyw** [reply]
umm... yes?

**Comment by vitalijalbu** [comment]
Also interested, would be great having like a theme tokens config file (JSON probably), **small**, **medium** (**default**) and **large**.... so all compionents will fit that theme...

**Comment by rasul-chopurov** [comment]
how to use searchable is custom view ?
 View::make('tables.cards.course-card')
 using and add to table ->searchable not work

**Reply by danharrin** [reply]
You need to pass the database column names to search into `searchable()`

**Reply by rasul-chopurov** [reply]
in fil doc, "For the global search input to show, at least one column in the table needs to be searchable()."
I am not using columns, use only custom view

**Reply by danharrin** [reply]
You will need to pass ->columns() into the view with searchable() on those, even if you do not plan to render them

**Reply by rasul-chopurov** [reply]
`TextColumn::make('name')->label('Kursun adı')->searchable(),`
commented
how to send to > ? 
`  View::make('tables.cards.course-card')`

**Reply by danharrin** [reply]
`View::make('tables.cards.course-card')->columns([TextColumn::make('name')->searchable()])`

**Reply by rasul-chopurov** [reply]
Method Filament\Tables\Columns\Layout\View::columns does not exist.

**Reply by danharrin** [reply]
Apologies, `->schema()`

**Reply by rasul-chopurov** [reply]
you saved my life!
